server/api/budget.mjs

```diff
@@ -0,0 +1,37 @@
+import { mongooseConnect } from "@/lib/mongoose";
+import { Budget } from "@/models/Budget";
+
+export default async function handle(req, res) {
+    const {method} = req; 
+    await mongooseConnect();
+
+
+    if ( method === 'GET') {
+        if (req.query?.id) {
+            res.json(await Budget.findOne({_id:req.query.id}));
+        } else { 
+            res.json(await Budget.find());
+        }
+    }
+
+    if (method === 'POST') {
+        const {budgetName,goalAmount,description,goalDate,status} = req.body;
```

Comment:
Again, can we guarantee this exists, or do we just pray it always is there? What happens if body does not exist 🤔